Source/WebCore/page/mac/ImageOverlayControllerMac.mm

```diff
@@ -208,34 +208,33 @@
         return;
     }
 
-    auto shadowHost = elementUnderMouse->shadowHost();
-    if (!is<HTMLElement>(shadowHost)) {
+    RefPtr imageOverlayHost = dynamicDowncast<HTMLElement>(elementUnderMouse->shadowHost());
+    if (!imageOverlayHost) {
         ASSERT_NOT_REACHED();
         m_hostElementForDataDetectors = nullptr;
         uninstallPageOverlayIfNeeded();
         return;
     }
 
-    Ref imageOverlayHost = downcast<HTMLElement>(*shadowHost);
-    if (!ImageOverlay::hasOverlay(imageOverlayHost.get())) {
+    if (!ImageOverlay::hasOverlay(*imageOverlayHost)) {
         ASSERT_NOT_REACHED();
         m_hostElementForDataDetectors = nullptr;
         uninstallPageOverlayIfNeeded();
         return;
     }
 
-    if (m_hostElementForDataDetectors == imageOverlayHost.ptr())
+    if (m_hostElementForDataDetectors == imageOverlayHost.get())
```

Comment:
Doesn’t this `==` work without the `get()`?

Reply:
No, the left side is a WeakPtr.